smartcar-auth/src/main/java/com/smartcar/sdk/SmartcarResponse.java

```diff
@@ -21,28 +21,86 @@
 package com.smartcar.sdk;
 
 /**
- * Class that encompasses the response code and error message, if any.
+ * A class that creates the response from Smartcar Connect.
  */
 public class SmartcarResponse {
     private String code;
+    private String error;
     private String message;
     private String state;
+    private VehicleInfo vehicleInfo;
 
-    public SmartcarResponse(String code, String message, String state) {
-        this.code = code;
-        this.message = message;
-        this.state = state;
+    /**
+     * Assigns properties on the SmartcarResponse object.
+     *
+     * @param builder the builder to obtain the properties from
+     */
+    private SmartcarResponse(Builder builder) {
+        this.code = builder.code;
+        this.error = builder.error;
+        this.message = builder.message;
+        this.state = builder.state;
+        this.vehicleInfo = builder.vehicleInfo;
     }
 
     public String getCode() {
-        return code;
+        return this.code;
+    }
+
+    public String getError() {
+        return this.error;
     }
 
     public String getMessage() {
```

Comment:
Pls change this to `getErrorDescription`

Reply:
changed